src/Microsoft.HttpRepl.Tests/HttpStateTests.cs

```diff
@@ -49,6 +53,128 @@ public void GetPathString_MultiplePathSections_CorrectString()
             Assert.Equal(expected, result);
         }
 
+        [Fact]
+        public void GetApplicableContentTypes_NoStructure_ReturnsNull()
+        {
+            HttpState httpState = SetupHttpState();
+
+            httpState.BaseAddress = new Uri("https://localhost/");
+            httpState.Structure = null;
+
+            IEnumerable<string> result = httpState.GetApplicableContentTypes(null, string.Empty);
+
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void GetApplicableContentTypes_NoMethod_ReturnsAll()
+        {
+            HttpState httpState = SetupHttpState();
+
+            DirectoryStructure directoryStructure = new DirectoryStructure(null);
+            RequestInfo requestInfo = new RequestInfo();
+            requestInfo.SetRequestBody("GET", "application/json", "");
+            requestInfo.SetRequestBody("PUT", "application/xml", "");
+            directoryStructure.RequestInfo = requestInfo;
+
+            httpState.BaseAddress = new Uri("https://localhost/");
+            httpState.Structure = directoryStructure;
+
+            IEnumerable<string> result = httpState.GetApplicableContentTypes(null, "");
+
+            Assert.NotNull(result);
+
+            List<string> resultList = result.ToList();
```

Comment:
nit: same here. Use result directly

Reply:
That one was sort of on purpose - because you can't check the length of an IEnumerable directly and I wanted to make sure it was exactly 2, not just that it had the two I was looking for. But I changed it and used the extension method. For that size, no perf reason not to. So that's changed in a702d3c